drivers/interrupt_controller/Kconfig.plic

```diff
@@ -10,9 +10,3 @@ config PLIC
 	help
 	  Platform Level Interrupt Controller provides support
 	  for external interrupt lines defined by the RISC-V SoC.
-
-config PLIC_SUPPORTS_EDGE_IRQ
```

Comment:
It isn't clear to me from [the spec](https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc) whether support for level and edge triggers is optional. So from that perspective, it makes sense to remove this option.